src/_gettsim/transfers/unterhaltsvors.py

```diff
@@ -68,56 +103,73 @@ def unterhaltsvors_m(  # noqa: PLR0913
     if (
         out > 0
         and (alter >= unterhaltsvors_params["altersgrenze_mindesteinkommen"])
-        and (unterhaltsvorschuss_eink_m_tu < unterhaltsvors_params["mindesteinkommen"])
+        and (not _unterhaltsvorschuss_eink_above_income_threshold_fg)
```

Comment:
This is just an approximation of the legal requirements because the income check is not really on the `fg` level:

Conceptually, we need to check whether the parent that receives Unterhaltsvorschuss satisfy an income threshold. This parental-level boolean should then be used on the child-level to check whether the child is eligible for Unterhaltsvorschuss. This is then again aggregated to the parent level to calculate `unterhaltsvors_m`.

I think we are not able to handle this case yet (without introducing a new input variable besides `p_id_unterhaltsvors_empf`).

Reply:
I'm afraid I don't follow the second paragraph just yet.
- How does that back and forth work?
- Is the check at the parent level really on the individual income? That is, if my `einstandspartner` is a millionaire, doesn't her income count?

Reply:
I think the problem is that we have pointers from children to parents, but not from parents to children. So, my point is that we cannot properly depict this minimum income check in this PR. Using `fg` as a grouping works as long as no child earns more than the minimum income threshold.

> And can't we determine p_id_unterhaltsvors_empf endogenously based on kindesunterhalt and parental ids?

Yes that's true, forgot about that.

Reply:
Lets see whether I understood correctly. We have

| p_id | fg_id | alter | p_id_unterhaltsvors_empf | einkommen_m | unterhalt |
| ---- | ----- | ----- | ------------------------ | ----------- | --------- |
| 0    | 0     | 35    | -1                       | 100         | 0         |
| 1    | 0     | 15    | 0                        | 0           | 0         |
| 2    | 1     | 35    | -1                       | 1000        | 0         |
| 3    | 1     | 15    | 2                        | 0           | 0         |
| 4    | 0     | 13    | 2                        | 0           | 500       |

and want to get at:

| p_id | unterhaltsvors_kind |
| ---- | ------------------- |
| 0    | 0                   |
| 1    | 0                   |
| 2    | 0                   |
| 3    | 250                 |
| 4    | 0                   |

because first parent does not qualify for income check and last kid receives too much Unterhalt.

The problem is, of course, that if we work on row-based data and vectorize, we can't get at: *(using pandas syntax)*

```py
data.loc["p_id_unterhaltsvors_empf", "einkommen"]
```

However, we could create a variables like this upfront with the whole aggregation stuff. In this case, we are not aggregating by any values, but we are only using pointers (this is the unique end of the m:1 relationship between parents and children). 

*In case this is correct:* This does not seem hard to do in case we don't have the machinery just yet. @lars-reimann, could you have a look at that? Might be worth doing a separate PR for that, using the above as the input and the intermediate variable:

| p_id | _unterhaltsvors_empf_einkommen_m |
| ---- | ------------------- |
| 0    | nan                   |
| 1    | 100.0                   |
| 2    | nan                   |
| 3    | 1000.0                 |
| 4    | 1000.0                   |

as the output of a test?

Reply:
Yes, that's exactly the problem.

Reply:
@lars-reimann, this probably got buried in the flurry of Marvin's activity -- could you have a look at my suggestion two messages above? I am thinking of a function `get_value_from_other_p_id(variable, others_p_id)` or so.

Reply:
Will be fixed in #708